app/presenters/document_export_presenter.rb

```diff
@@ -12,60 +12,127 @@ def as_json
 
   def editions
     model.editions.map do |edition|
-      edition_associations(edition)
+      {
+        edition: edition,
+        government: edition.government,
+        whitehall_admin_links: resolve_whitehall_admin_links(edition),
+        associations: {
+          attachments: complete_attachments(edition),
+          authors: stripped_authors(edition),
+          edition_organisations: complete_organisations(edition),
+          edition_policies: stripped_edition_policies(edition),
+          images: complete_images(edition),
+          last_author: stripped_last_author(edition),
+          topical_events: strippped_topical_events(edition),
+          translations: complete_translations(edition),
+          specialist_sectors: stripped_specialist_sectors(edition),
+        },
+      }.merge(provide_doctype_information(edition))
     end
   end
 
-  def edition_associations(edition)
-    output = {
-               edition: edition,
-               associations: {},
-             }
-
-    associations = edition.class.reflect_on_all_associations.map(&:name)
-    associations.each do |association|
-      if association == :images
-        edition_images = edition.public_send(association)
-        output[:associations][association] = complete_images_hash(edition_images)
-      elsif association == :attachments
-        edition_attachments = edition.public_send(association)
-        output[:associations][association] = complete_attachments_hash(edition_attachments)
-      else
-        output[:associations][association] = edition.public_send(association)
-      end
+  def stripped_last_author(edition)
+    author = edition.try(:last_author)
+    return {} unless author
+
+    { id: author.id, uid: author.uid }
+  end
+
+  def strippped_topical_events(edition)
+    topical_events = edition.try(:topical_events)
+    return [] unless topical_events
+
+    topical_events.map do |te|
+      { id: te.id, content_id: te.content_id }
     end
+  end
 
-    provide_doctype_information(edition, output)
-    output[:government] = edition.government
-    output[:whitehall_admin_links] = resolve_whitehall_admin_links(edition.body)
-    if edition.withdrawn?
-      output[:whitehall_admin_links].concat(resolve_whitehall_admin_links(edition.unpublishing.explanation))
+  def complete_translations(edition)
+    translations = edition.try(:translations)
+    return [] unless translations
+
+    translations.map do |translation|
+      translation.attributes.except("edition_id").merge(
```

Comment:
Nice use of `except` - you're right, we don't need that field 👍However, think you can use the `:edition_id` symbol instead here.

Reply:
Hmm don't think so as `translation.attributes` returns string keys, you could if you called something like hash_with_indifferent_access on it though - https://api.rubyonrails.org/classes/ActiveSupport/HashWithIndifferentAccess.html.

Reply:
Yeah I'm using that method on my PR. Hmm on second thoughts that would just make this weightier with little added value. Fine to ignore!